Dockerfile.refactored

```diff
@@ -0,0 +1,255 @@
+ENV LANGUAGE $LOCALE
+ENV LC_ALL $LOCALE
+ENV LANG $LOCALE
+
+# Install general APT packages
+RUN apt-get --quiet --yes install  \
+    language-pack-en \
+    wget \
+    unzip \
+    git \
+    locales \
+    s4cmd \
+    logrotate \
+    supervisor \
+    cron \
+    syslog-ng \
+    rsync \
+    curl \
+    dnsutils \
+    busybox \
+    tzdata \
+    uuid-dev
+
+# Install services
+RUN apt-get --quiet --yes install  \
+    nginx \
+    apache2 \
+    apache2-utils \
+    apache2-dev \
+    libapache2-mod-xsendfile \
+    postgresql \
+    redis-server \
+    rabbitmq-server \
+    unixodbc-dev
+
+# Install build tools
+RUN apt-get --quiet --yes install  \
+    gcc \
+    make \
+    autoconf \
+    automake \
+    build-essential
+
+# Install programming languages
+RUN apt-get --quiet --yes install  \
+    perl \
+    libinline-perl \
+    libparallel-forkmanager-perl \
+    npm \
+    python3-passlib \
+    python3-pip \
+    python3-venv \
+    gfortran
+
+# Install text and image manipulation tools
+RUN apt-get --quiet --yes install  \
+    imagemagick \
+    pdftk \
+    pacpl \
+    pandoc \
+    poppler-utils \
+    ffmpeg \
+    tesseract-ocr-all \
+    fonts-ebgaramond-extra \
+    ghostscript \
+    fonts-liberation \
+    cm-super \
+    qpdf \
+    wamerican \
+    libreoffice
+
+# Install various libraries
+RUN apt-get --quiet --yes install  \
+    libgdbm-dev \
+    libdb5.3-dev \
+    libbz2-dev \
+    libexpat1-dev \
+    liblzma-dev \
+    libffi-dev \
+    libaugeas0 \
+    zlib1g-dev \
+    libncurses5-dev \
+    libncursesw5-dev \
+    libreadline-dev \
+    libsqlite3-dev \
+    liblapack-dev \
+    libblas-dev \
+    libjpeg-dev \
+    libpq-dev \
+    libxml2 \
+    libxslt1.1 \
+    libu2f-udev \
+    libvulkan1 \
+    libxml2-dev \
+    libxslt1-dev \
+    libcurl4-openssl-dev \
+    libssl-dev \
+    libtool \
+    libtool-bin \
+    libsvm3 \
+    libsvm-dev \
+    liblinear4 \
+    liblinear-dev \
+    libzbar-dev \
+    libzbar0 \
+    libgs-dev \
+    default-libmysqlclient-dev \
+    libgmp-dev \
+    libsasl2-dev \
+    libldap2-dev \
+    exim4-daemon-heavy \
+    libaudio-flac-header-perl \
+    libaudio-musepack-perl \
+    libmp3-tag-perl \
+    libogg-vorbis-header-pureperl-perl \
+    libvorbis-dev \
+    libcddb-perl \
+    libcddb-get-perl \
+    libmp3-tag-perl \
+    libaudio-scan-perl \
+    libaudio-flac-header-perl \
+    libtesseract-dev
+
+# This installs correctly if done before the rext of the texlive installs.
+RUN apt-get --quiet --yes install texlive-latex-extra
+
+# Installing Tex stuff
+RUN apt-get --quiet --yes install \
+    texlive \
+    texlive-xetex \
+    texlive-latex-recommended \
+    texlive-luatex \
+    texlive-latex-extra \
+    texlive-font-utils \
+    texlive-lang-cyrillic \
+    texlive-lang-french \
+    texlive-lang-italian \
+    texlive-lang-portuguese \
+    texlive-lang-german \
+    texlive-lang-european \
+    texlive-lang-spanish
+
+# Remove unwanted packages
```

Comment:
Unsolicited comment, but using a separate `RUN` command to remove files doesn't actually make the resulting image smaller, since each `RUN` command in a dockerfile creates a new layer; so all of these removed files are still saved in the image.

[Docker itself recommends that you minimize the number of layers](https://docs.docker.com/build/cache/#minimize-the-number-of-layers) in a docker image, something that the original `Dockerfile` was doing.

Reply:
I tried to perform all the steps that were in the original.  Remove was there so I left it.

The original Dockerfile takes the "minimize your layers" to an extreme.  Just like putting creating an apt-get install line for each package would be excessive in the other direction.  The "code" needs to be readable and troubleshoot-able.  I'm not saying my balance of layers is perfect but I think it is better than all in one.